src/rgw/rgw_data_sync.cc

```diff
@@ -1014,7 +1014,7 @@ class RGWDataSyncSingleEntryCR : public RGWCoroutine {
         if (error_repo && !error_repo->append(raw_key)) {
           ldout(sync_env->store->ctx(), 0) << "ERROR: failed to log sync failure in error repo: retcode=" << retcode << dendl;
         }
-      } else if (error_repo && remove_from_repo) {
+      } else if (error_repo && remove_from_repo && (retcode >= 0)) {
```

Comment:
maybe when -ENOENT, we should not exit RGWBucketShardIncrementalSyncCR.
if remove here, sync maybe not finish,and there is no way to trigger bucket sync.